classes/sota_sanity.bbclass

```diff
@@ -10,6 +10,55 @@ def sota_check_required_variables(status, d):
         if not d.getVar(var):
             status.addresult("%s should be set in your local.conf.\n" % var)
 
+def sota_check_variables_validity(status, d):
+    import re
+    import os.path
+
+# Translate old provisioning recipe names into the new versions.
```

Comment:
If I put the `Translate old provisioning recipe names` bits here, and test with an old name, the build will fail. The warning message showed the translation should have been executed, once. So, is it that the translation should be executed more than once? Here is the bitbake log:
```
WARNING: aktualizr-auto-prov is deprecated. Please use aktualizr-shared-prov instead.
Time: 0:00:03
Sstate summary: Wanted 2 Found 0 Missed 2 Current 1246 (0% match, 99% complete)
NOTE: Executing Tasks
NOTE: Setscene tasks completed
ERROR: core-image-minimal-1.0-r0 do_rootfs: Could not invoke dnf. Command '/home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/recipe-sysroot-native/usr/bin/dnf -v --rpmverbosity=info -y -c /home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/rootfs/etc/dnf/dnf.conf --setopt=reposdir=/home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/rootfs/etc/yum.repos.d --installroot=/home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/rootfs --setopt=logdir=/home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/temp --repofrompath=oe-repo,/home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/oe-rootfs-repo --nogpgcheck install aktualizr aktualizr-auto-prov aktualizr-resource-control os-release ostree packagegroup-core-boot run-postinsts systemd-journald-persistent virtual/network-configuration' returned 1:
DNF version: 4.2.2
cachedir: /home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/rootfs/var/cache/dnf
Added oe-repo repo from /home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/oe-rootfs-repo
repo: using cache for: oe-repo
not found other for: 
not found modules for: 
not found deltainfo for: 
not found updateinfo for: 
oe-repo: using metadata from Wed 20 Nov 2019 11:06:39 AM UTC.
Last metadata expiration check: 0:00:01 ago on Wed 20 Nov 2019 11:06:39 AM UTC.
No module defaults found
No match for argument: aktualizr-auto-prov
Error: Unable to find a match

ERROR: Logfile of failure stored in: /home/liping/meta-updater-repo/master/build/tmp/work/qemux86_64-poky-linux/core-image-minimal/1.0-r0/temp/log.do_rootfs.24967
ERROR: Task (/home/liping/meta-updater-repo/master/poky/meta/recipes-core/images/core-image-minimal.bb:do_rootfs) failed with exit code '1'


```

Reply:
I think I see the problem. Unfortunately, the provisioning translation logic might have to be in `sota.bbclass` to apply everywhere that is used.

However, it's been about six months since we introduced the new names, and I wonder if it's time to deprecate the old names entirely. So instead of translating them, we should print a warning and abort. Then we can get rid of that python script, not have to check and translate recipe names everywhere, and only do the check for the old names once here in a sanity function.

Reply:
I prefer this solution.

Reply:
Just to be clear, by "this solution" do you mean my proposal to stop translating and just abort instead?

Reply:
Yes, that’s what I meant.

Reply:
Check `SOTA_CLIENT_PROV` in `sota_sanity.bbclass` would cause os-selftest failure, even if I put the translating bits back to `sota.bbclass`. But I can't find where the deprecated name is used?

Reply:
Looks `SOTA_CLIENT_PROV` may also include more than one option.
Setting `SOTA_CLIENT_PROV`  to an old name in `local.conf` would cause bitbake failure if the translation is in` sota_sanity.bbclass`. It's maybe a lot easier just stop translating and abort. Is there any other concerns?

Reply:
> Looks `SOTA_CLIENT_PROV` may also include more than one option.

That isn't valid; it wouldn't make sense.

> It's maybe a lot easier just stop translating and abort.

Yes, I agree, we shouldn't translate anymore. If someone tries to use the old names, we can suggest the new name, but I think that's enough at this point.

Reply:
`self.append_config('SOTA_CLIENT_PROV = " aktualizr-shared-prov "') `
（From `lib/oeqa/selftest/cases/updater_qemux86_64.py` `lib/oeqa/selftest/cases/updater_minnowboard.py` `lib/oeqa/selftest/cases/updater_raspberrypi.py` ) 
The blank space behind `aktualizr-shared-prov` seems to be the reason to cause oe-selftest failure.

Reply:
Wow, good catch. So I guess then you either have to strip the whitespace out or just search if the expected values are in the value (instead of equivalent).